### PR TITLE
Problem: no decision record for crc21 standard and crc20 token migration

### PR DESCRIPTION
**Pull request by thomas-nguy**
👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

Add ADR-006

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

**Review by tomtau — COMMENTED** [review]
I don't think many dApps were planning CRC20 in advance, many of them may not even comply with CRC20 (i.e. they are plain ERC20), so likely the second solution is unavoidable

**Comment by thomas-nguy** [comment]
seems like the general opinion tends forward solution #2.

I will update some spec for CRC21 and update this afterward

**Comment by thomas-nguy** [comment]
@CeruleanAtMonaco @calvinaco any comments about the ADR?

If we opt for solution 2, it means that a wrapped contract will be needed for CRC20 that want to be bridge to ethereum (or other external evm chains)

**Comment by calvinaco** [comment]
> @CeruleanAtMonaco @calvinaco any comments about the ADR?
> 
> If we opt for solution 2, it means that a wrapped contract will be needed for CRC20 that want to be bridge to ethereum (or other external evm chains)

I am also skeptical on solution 1. Upgrading bytecode after a contract is deployed could be opening a pandora box. It would be breaking user assumptions on the immutability of smart contract code.

**Comment by yihuang** [comment]
> I am also skeptical on solution 1. Upgrading bytecode after a contract is deployed could be opening a pandora box. It would be breaking user assumptions on the immutability of smart contract code.

It's only done together with chain upgrade, and only for limited contracts which are issued by crypto-com.
The chain upgrade needs to go through the governance voting to gain legitimacy.

**Comment by calvinaco** [comment]
> > I am also skeptical on solution 1. Upgrading bytecode after a contract is deployed could be opening a pandora box. It would be breaking user assumptions on the immutability of smart contract code.
> 
> It's only done together with chain upgrade, and only for limited contracts which are issued by crypto-com. The chain upgrade needs to go through the governance voting to gain legitimacy.

There could be external dApp and project who needs to upgrade as well. I agree the governance voting can make sure the community is aware of the bytecode update. Also the bytecode is verifiable as well. But this solution doesn't seems to be "scalable" IMO.

**Comment by thomas-nguy** [comment]
According to @lezzokafka there are more than 1000 CRC20 contract deployed on cronos, and some of them want to be bridge to Ethereum. Upgrading through governance does not seem scalable indeed.

I think most of them will have to use a contract wrapper except if there is another solution (cc @FrancoCRO)

**Comment by yihuang** [comment]
> > > I am also skeptical on solution 1. Upgrading bytecode after a contract is deployed could be opening a pandora box. It would be breaking user assumptions on the immutability of smart contract code.
> > 
> > 
> > It's only done together with chain upgrade, and only for limited contracts which are issued by crypto-com. The chain upgrade needs to go through the governance voting to gain legitimacy.
> 
> There could be external dApp and project who needs to upgrade as well. I agree the governance voting can make sure the community is aware of the bytecode update. Also the bytecode is verifiable as well. But this solution doesn't seems to be "scalable" IMO.

Solution 1 is just try to do sth on chain side to ease the pain of upgrading for certain token contracts. solution 2 is to do nothing on the chain side, just rely on dapp side to do either contract upgrade or token wrapping.

**Comment by thomas-nguy** [comment]
@calvinaco @yihuang @tomtau updated the decision record

File: docs/architecture/README.md [hunk 1]
```diff
@@ -32,3 +32,4 @@ To suggest an ADR, please make use of the [ADR template](./adr-template.md) prov
 | [003](./adr-003.md) | Add Fee Market Module | Accepted |
 | [004](./adr-004.md) | Tokens conversion in Cronos | Accepted |
 | [005](./adr-005.md) | Cross-chain Validation for Gravity Bridge | Rejected |
+| [006](./adr-006.md) | Migrating CRC20 contract to CRC21 standard | Proposed |
```
Comment (tomtau) [inline docs/architecture/README.md:35]:
```suggestion
| [006](./adr-006.md) | Migrating CRC20 contract to CRC21 standard | Rejected |
```

**Review by tomtau — APPROVED** [review]

**Review by yihuang — APPROVED** [review]

**Pull request merged**